source/_drafts/Feature-Spotlight-Middleware.md

```diff
@@ -0,0 +1,187 @@
+---
+title: 'Feature Spotlight: Middleware'
+author:
+  name: Gary Katsevman
+  github: gkatsev
+tags:
+  - middleware
+  - video.js 6.0
+---
+
+Middleware is one of the cool new features that is coming to Video.js in version 6.0
+
+With middleware, you are now able to interact with and change how the player and the tech talk to each other.
+The tech is Video.js's abstraction from the player, separating the player API from the playback technology.
+With techs, we can plug things like a Flash fallback or a Youtube embed into Video.js without changing
+the external API or the look-and-feel of the player.
+
+A lot of Video.js users may be familiar with middleware from projects like Express.
+Video.js middleware isn't that different from those.
+In both cases you register your middleware against a particular route to call down the chain when the route is triggered.
+In Express, routes are based on the url paths.
+In Video.js, these routes are based on the video MIME type.
+And, like Express, there are "star" (`*`) middleware that match all routes.
+
+There are two important pieces to be aware of with middleware:
+1. dynamic source handling
+2. intercepting the player and tech interactions.
+
+With the dynamic source handling, you could load video with a custom type and source and resolve it asynchronously.
+A good example for this is a video catalog system.
+The page can be rendered with a specific catalog ID and a special MIME type, like so:
+```html
+<video controls class="video-js">
+  <source src="123" type="video/my-catalog">
+</video>
+```
+Then, you could register a middleware for that route -- the type `video/my-catalog`.
+```js
+// middleware methods get the player instance as an argument
+videojs.use('video/my-catalog', function(player) {
+
+  // middleware are expected to return an object with the methods on it.
+  // It can be a plain object or an instance of something.
+  return {
+
+    // setSource allows you to tell Video.js whether you're going to be handling the source or not
+    setSource(srcObj, next) {
+      const id = srcObj.src;
+
+      videojs.xhr({
+        uri: '/getVideo?id=' + id
+      }, function(err, res, body) {
+
+        // pass null as the first argument to say everything is going fine and we can handle it.
+        next(null, {
+          src: body.sourceUrl,
+          type: body.sourceType
+        })
+      });
+    }
+  };
+});
+```
+Then, when Video.js initializes, it'll go through and call the middleware that are set for `video/my-catalog`.
+
+Server Side Ad Insertion (SSAI) is a great fit for middleware. It showcases the ability to intercept the play and tech interactions.
```

Comment:
Might be good to make this a heading as well. "Server Side Ad Insertion (SSAI) and `*` middleware"